src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -234,6 +234,62 @@ public decimal GetDecimal()
             return value;
         }
 
+        /// <summary>
+        /// Reads the next JSON token value from the source and parses it to a <see cref="DateTime"/>.
+        /// Returns the value if the entire UTF-8 encoded token value can be successfully parsed to a <see cref="DateTime"/>
+        /// value.
+        /// Throws exceptions otherwise.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of a JSON token that is not a <see cref="JsonTokenType.String"/>.
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <exception cref="FormatException">
+        /// Thrown if the JSON token value is of unsupported or incorrect ISO 8601 format.
+        /// </exception>
+        public DateTime GetDateTime()
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
+            }
+
+            if (!TryGetDateTime(out DateTime value))
+            {
+                throw ThrowHelper.GetFormatException(DateType.DateTime);
+            }
+
+            return value;
+        }
+
+        /// <summary>
+        /// Reads the next JSON token value from the source and parses it to a <see cref="DateTimeOffset"/>.
+        /// Returns the value if the entire UTF-8 encoded token value can be successfully parsed to a <see cref="DateTimeOffset"/>
+        /// value.
+        /// Throws exceptions otherwise.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of a JSON token that is not a <see cref="JsonTokenType.String"/>.
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <exception cref="FormatException">
+        /// Thrown if the JSON token value is of unsupported or incorrect ISO 8601 format.
+        /// </exception>
+        public DateTimeOffset GetDateTimeOffset()
+        {
+            if (TokenType != JsonTokenType.String)
```

Comment:
Isn't this check being duplicated here (since `TryGetDateTimeOffset` does it already). If so, I would remove it.